corehq/form_processor/management/commands/populate_missing_delete_on_values.py

```diff
@@ -0,0 +1,39 @@
+from django.db.models import Q
+from django.core.management.base import BaseCommand
+
+from corehq.form_processor.models import XFormInstance
+from corehq.sql_db.util import paginate_query_across_partitioned_databases
+
+from dimagi.utils.chunked import chunked
+
+
+class Command(BaseCommand):
+    help = "One time use to populate missing XFormInstance.deleted_on values"
+
+    def handle(self, **options):
+        chunk_size = 100
+        old_deleted_state = 64
+
+        def iter_all_ids():
+            rows = paginate_query_across_partitioned_databases(
+                XFormInstance,
+                Q(deleted_on__isnull=True, state__gte=old_deleted_state),
+                values=['form_id'],
+                query_size=chunk_size,
+            )
+            for r in rows:
+                yield r[0]
+
+        form_ids = iter_all_ids()
+        for chunk in chunked(form_ids, chunk_size, list):
+            with XFormInstance.get_plproxy_cursor() as cursor:
```

Comment:
This cursor is pointing to the 'proxy' database which has no data on it. That's why this isn't working on staging. You'll need to use `split_list_by_db_partition` to split the list of IDs by database and then do a query per DB:

```python
for db, db_chunk in split_list_by_db_partition(chunk):
    with XFormInstance.get_cursor_for_partition_db(db) as cursor:
        ...
```

Reply:
ahhh yea that'll do it. Now I'm curious though - what is the purpose the proxy database? Is it just for testing?

Reply:
If you are calling one of the PlProxy functions you need to do that on the proxy database